### PR TITLE
docs: fix index wrapper inconsistencies

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/3632

Also removes a few sections about intermediate state roots b/c those haven't been implemented (and potentially will never be). Note: there are other references to intermediate state roots in the specs that we need to resolve so we still need to do https://github.com/celestiaorg/celestia-app/issues/3113

### CODE REVIEW

File: specs/src/specs/data_structures.md [hunk 1]
```diff
@@ -370,46 +369,21 @@ The blob share commitment rules may introduce empty shares that do not belong to
...
+| `tx`            | `bytes`    | Actual transaction.                                                                                                                                    |
+| `share_indexes` | `[]uint32` | Share indexes (in row-major order) of the first share for each blob this transaction pays for. Needed for light verification of proper blob inclusion. |
+| `type_id`       | `string`   | Type ID of the IndexWrapper transaction type. Always set to `"INDX"`                                                                                   |
 
-#### IntermediateStateRoot
-
-| name   | type                      | description                                                                              |
-|--------|---------------------------|------------------------------------------------------------------------------------------|
-| `root` | [HashDigest](#hashdigest) | Root of intermediate state, which is composed of the global state and the validator set. |
+### PayForBlobData
```
Comment (cmwaters) [inline specs/src/specs/data_structures.md:386]: Shouldn't the `IndexWrapper` go beneath the pay for blob data?

File: specs/src/specs/data_structures.md [hunk 2]
```diff
@@ -370,46 +369,21 @@ The blob share commitment rules may introduce empty shares that do not belong to
...
+| name            | type       | description                                                                                                                                            |
+|-----------------|------------|--------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `tx`            | `bytes`    | Actual transaction.                                                                                                                                    |
+| `share_indexes` | `[]uint32` | Share indexes (in row-major order) of the first share for each blob this transaction pays for. Needed for light verification of proper blob inclusion. |
+| `type_id`       | `string`   | Type ID of the IndexWrapper transaction type. Always set to `"INDX"`                                                                                   |
```
Comment (cmwaters) [inline specs/src/specs/data_structures.md:384]: I would maybe specify that this is used for encoding purposes to help ensure uniqueness

File: specs/src/specs/data_structures.md [hunk 3]
```diff
@@ -72,12 +72,11 @@ Implementations can prune rows containing only [tail padding](./consensus.md#res
...
+| name                        | type                                                    | description                                                                                                           |
+|-----------------------------|---------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------|
+| `transactions`              | [Transaction](#transaction)                             | Transactions are ordinary Cosmos SDK transactions. For example: they may modify the validator set and token balances. |
+| `payForBlobData`            | [PayForBlobData](#payforblobdata)                       | PayForBlob data. Transactions that pay for blobs to be included.                                                      |
+| `blobData`                  | [BlobData](#blobdata)                                   | Blob data. Blobs are app data.                                                                                        |
```
Comment (cmwaters) [inline specs/src/specs/data_structures.md:79]: ```Blobs are app data```

This feels like an incorrect description

Comment (rootulp) [reply specs/src/specs/data_structures.md:79]: The specs use "app data" to refer to data published by applications on top of Celestia (like roll-ups).

File: specs/src/specs/data_structures.md (hunk 1 above)
Comment (rootulp) [reply specs/src/specs/data_structures.md]: 2652b1021f8af9b81383699f20664aa6d34401c7

File: specs/src/specs/data_structures.md (hunk 3 above)
Comment (rootulp) [reply specs/src/specs/data_structures.md]: dc14f8b027aed885cbcd4067632eaab3ae5980b7

File: specs/src/specs/data_structures.md (hunk 2 above)
Comment (rootulp) [reply specs/src/specs/data_structures.md]: 7e11c5ce5af4f9d58bdfd13251f4cfc9d462648d

**Review by cmwaters — APPROVED** [review]
This is a good start. Looking through the specs there seems to be a lot of aspects that are outdated which makes me wonder if it's worth doing an entire pass through and trying to update them ahead of the v2 upgrade

**Comment by rootulp** [comment]
Agreed there's a lot in the specs that aren't accurate with respect to the current implementation. We have issues for some of it: https://github.com/celestiaorg/celestia-app/issues?q=is%3Aissue+is%3Aopen+label%3Aspecs

**Review by cmwaters — APPROVED** [review]

**Review by ninabarbakadze — APPROVED** [review]

**Pull request merged**